pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -132,6 +132,25 @@ public boolean equals(Object obj) {
     }
 
     public void publishMessage(long producerId, long sequenceId, ByteBuf headersAndPayload, long batchSize) {
+        beforePublish(producerId, sequenceId, headersAndPayload, batchSize);
+        publishMessageToTopic(headersAndPayload, sequenceId, batchSize);
+    }
+
+    public void publishMessage(long producerId, long sequenceId, long lastSequenceId,
+           ByteBuf headersAndPayload, long batchSize) {
+        if (sequenceId > lastSequenceId) {
```

Comment:
`sequenceId` => `lowestSequenceId`
`lastSequenceId` => `highestSequenceId`

It is very confusing using `sequenceId` and `lastSequenceId`.